modules/delegates.js

```diff
@@ -316,7 +316,9 @@ __private.checkDelegates = function (publicKey, votes, state, cb) {
 			}
 
 			if (math === '+' && (delegates != null && delegates.indexOf(publicKey) !== -1)) {
-				return setImmediate(cb, 'Failed to add vote, account has already voted for this delegate');
+                               modules.accounts.getAccount({publicKey: publicKey, isDelegate: 1}, function(err, acc) {
+                                 return setImmediate(cb, err ? err : "Failed to add vote, account already voted for " + acc.username)
+                               })
 			}
 
 			if (math === '-' && (delegates === null || delegates.indexOf(publicKey) === -1)) {
```

Comment:
We should do something similar here as well. Like, return an error saying: 'Failed to remove vote, account has already voted for' + acc.username.